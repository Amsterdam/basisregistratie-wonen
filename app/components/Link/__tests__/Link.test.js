import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import { ThemeProvider } from '@datapunt/asc-ui';
import 'jest-styled-components';
import Link from '..';

describe('Link', () => {
  afterEach(cleanup);

  it('matches the snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Link href="//some-url.com" label="Foo bar" />
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.getAttribute('href')).toBe('//some-url.com');
  });

  it('calls the onClick handler', () => {
    let someVar = 'a';

    const onClick = () => {
      someVar = 'b';
    };

    const { getByText } = render(
      <ThemeProvider>
        <Link href="//some-url.com" label="Bar baz" onClick={onClick} />
      </ThemeProvider>,
    );

    fireEvent(
      getByText('Bar baz'),
      new MouseEvent('click', {
        bubbles: true,
      }),
    );

    expect(someVar).toBe('b');
  });
});
