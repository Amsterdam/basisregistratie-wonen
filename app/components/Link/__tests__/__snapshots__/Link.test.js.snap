// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link matches the snapshot 1`] = `
.c0 {
  padding: 0 !important;
}

.c0:focus {
  outline: 4px solid #ffc90e !important;
}

.c0:hover,
.c0:focus {
  color: #ec0000;
  fill: #ec0000;
}

.c2 {
  vertical-align: baseline;
  margin-right: 5px;
}

.c1 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c1:focus {
  outline: none;
  background-color: #FEC813;
}

.c1:hover {
  color: #ec0000;
}

<a
  class="c0 c1"
  href="//some-url.com"
>
  <img
    class="c2"
    width="14"
  />
  <span
    class="linklabel"
  >
    Foo bar
  </span>
</a>
`;
